Content.Server/Administration/ServerAPI.cs

```diff
@@ -0,0 +1,795 @@
+        {
+            // There is no function to get a session by GUID, so we have to iterate over all sessions and check their GUIDs.
+            foreach (var player in _playerManager.Sessions)
+            {
+                if (player.UserId.UserId.ToString() == guid.ToString())
+                {
+                    return player;
+                }
+            }
+            return null;
+        });
+
+        if (session == null)
+        {
+            _sawmill.Info($"Attempted to kick player {guid} by {actor!.Name}({actor!.Guid}), but they were not found.");
+            await context.RespondAsync("Player not found", HttpStatusCode.NotFound);
+            return true;
+        }
+
+        var reasonSupplied = context.RequestHeaders.TryGetValue("Reason", out var reason);
+        if (!reasonSupplied)
+        {
+            reason = "No reason supplied";
+        }
+
+        reason += " (kicked by admin)";
+
+        _taskManager.RunOnMainThread(() =>
+        {
+            _netManager.DisconnectChannel(session.Channel, reason.ToString());
+        });
+        await context.RespondAsync("Success", HttpStatusCode.OK);
+        _sawmill.Info("Kicked player {0} ({1}) for {2} by {3}({4})", session.Name, session.UserId.UserId.ToString(), reason, actor!.Name, actor!.Guid);
+        return true;
+    }
+
+    /// <summary>
+    ///     Round restart/end
+    /// </summary>
+    private async Task<bool> ActionRoundStatus(IStatusHandlerContext context)
+    {
+        if (context.RequestMethod != HttpMethod.Post || context.Url!.AbsolutePath != "/admin/actions/round")
+        {
+            return false;
+        }
+
+        if (!CheckAccess(context))
+        {
+            await context.RespondErrorAsync(HttpStatusCode.Unauthorized);
+            return true;
+        }
+
+        if (!CheckActor(context, out var actor))
+        {
+            await context.RespondAsync("An actor is required to perform this action.", HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        // Not using body, because that's a stream and I don't want to deal with that
+        var actionSupplied = context.RequestHeaders.TryGetValue("Action", out var action);
+        if (!actionSupplied)
+        {
+            _sawmill.Info($"Attempted to {action} round without supplying an action by {actor!.Name}({actor!.Guid}).");
+            await context.RespondErrorAsync(HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        var ticker = _entitySystemManager.GetEntitySystem<GameTicker>();
+        var roundEndSystem = _entitySystemManager.GetEntitySystem<RoundEndSystem>();
+        switch (action)
+        {
+            case "start":
+                if (ticker.RunLevel != GameRunLevel.PreRoundLobby)
+                {
+                    await context.RespondAsync("Round already started", HttpStatusCode.BadRequest);
+                    _sawmill.Info("Forced round start failed: round already started");
```

Comment:
Errors like this should report something like `UnprocessableEntity` instead. `BadRequest` means the request is malformed, which this isn't.

Also these error codes seriously need a structured response so that `SS14.Admin` can properly display them instead of "unknown error".